src/hotspot/share/gc/parallel/psOldGen.hpp

```diff
@@ -44,6 +46,7 @@ class PSOldGen : public CHeapObj<mtGC> {
   // Performance Counters
   PSGenerationCounters*    _gen_counters;
   SpaceCounters*           _space_counters;
+  PretouchTask*            _pretouch;          // Used when old gen resized during scavenging.
```

Comment:
Thanks for your suggestion. It will be better if you can give some direction as G1GC also need similar fix during the expansion. I think G1 has array of regions to be touched during the expansion and that task needs to be shared across the threads. pretouch class can be put inside another class and that can be shared across the threads to synchronize the pretouch task and not sure whether that will be OK again. I explored this approach after your suggestion and thought to update after the feedback during the review process. Please suggest.